pkl-core/src/main/java/org/pkl/core/ast/type/TypeNode.java

```diff
@@ -2502,6 +2519,22 @@ protected boolean acceptTypeNode(TypeNodeConsumer consumer) {
     protected boolean isParametric() {
       return typeArgumentNodes.length > 0;
     }
+
+    // Note that mutating a frame's receiver and owner argument is very risky
+    // because any VmObject instantiated within the same root node execution
+    // holds a reference to (not immutable snapshot of) the frame
+    // via VmObjectLike.enclosingFrame.
+    // *Maybe* this works out for TypeAliasTypeNode because an object instantiated
+    // within a type constraint doesn't escape the constraint expression.
+    // If mutating receiver and owner can't be avoided, it would be safer
+    // to have VmObjectLike store them directly instead of storing enclosingFrame.
```

Comment:
> I see the value of aspirational naming, but then, ideally, we'd have a TODO comment and a linked Issue. If it stays aspirational beyond the point of us remembering, it becomes misleading.

I've now changed the method name to `isValueTypeKnownSubtypeOf`, at which point it's no longer aspirational.
The method already handles `ValueType` <: `Any` and `ValueType` <: `Unknown`.
Improving this further is part of (6) on my planned list of improvements.

Reply:
> Given this danger; why not make a synthetic extra frame, on top of the given frame, with the new owner/receiver?

Turns out this may be possible:
```java
var newArgs = frame.getArguments().clone();
newArgs [0] = newReceiver;
newArgs [1] = newOwner;
var newFrame = Truffle.getRuntime().createVirtualFrame(newArgs, frame.getDescriptor());
childNode.execute(newFrame);
```